api/class-wc-rest-dev-mailchimp-settings-controller.php

```diff
@@ -0,0 +1,307 @@
+
+	/**
+	 * MailChimp settings.
+	 *
+	 * @param  WP_REST_Request    $request    Request object.
+	 * @return WP_REST_Response   $response   Response data.
+	 */
+
+	/**
+	 * Register MailChimp settings routes
+	 */
+	public function register_routes() {
+		register_rest_route( $this->namespace, '/' . $this->rest_base, array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_settings' ),
+				'permission_callback' => array( $this, 'permissions_check' ),
+			)
+		) );
+		register_rest_route( $this->namespace, '/' . $this->rest_base . '/api_key', array(
+			array(
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => array( $this, 'update_api_key' ),
+				'permission_callback' => array( $this, 'permissions_check' ),
+			),
+		) );
+		register_rest_route( $this->namespace, '/' . $this->rest_base . '/store_info', array(
+			array(
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => array( $this, 'update_store_info' ),
+				'permission_callback' => array( $this, 'permissions_check' ),
+			),
+		'schema' => array( $this, 'get_store_info_schema' ),
+		) );
+		register_rest_route( $this->namespace, '/' . $this->rest_base . '/campaign_defaults', array(
+			array(
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => array( $this, 'update_campaign_defaults' ),
+				'permission_callback' => array( $this, 'permissions_check' ),
+			),
+		) );
+		register_rest_route( $this->namespace, '/' . $this->rest_base . '/newsletter_setting', array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_newsletter_settings' ),
+				'permission_callback' => array( $this, 'permissions_check' ),
+			),
+		) );
+		register_rest_route( $this->namespace, '/' . $this->rest_base . '/newsletter_setting', array(
+			array(
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => array( $this, 'update_newsletter_settings' ),
+				'permission_callback' => array( $this, 'permissions_check' ),
+			),
+		) );
+		register_rest_route( $this->namespace, '/' . $this->rest_base . '/sync', array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_sync_status' ),
+				'permission_callback' => array( $this, 'permissions_check' ),
+			),
+		) );
+		register_rest_route( $this->namespace, '/' . $this->rest_base . '/sync', array(
+			array(
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => array( $this, 'resync' ),
+				'permission_callback' => array( $this, 'permissions_check' ),
+			),
+		) );
+	}
+
+	/**
+	 * Makes sure the current user has access to WRITE the settings APIs.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 * @return WP_Error|boolean
+	 */
+	private function permissions_check( $request ) {
```

Comment:
This needs to be a `public` method so WP-API can properly use it.

`PHP Warning:  call_user_func() expects parameter 1 to be a valid callback, cannot access private method WC_REST_Dev_MailChimp_Settings_Controller::permissions_check()`